File: php/features.php
<?php
declare(strict_types=1);

// LongevityCompetition SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LongevityCompetitionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LongevityCompetitionBaseFeature();
            case "test":
                return new LongevityCompetitionTestFeature();
            default:
                return new LongevityCompetitionBaseFeature();
        }
    }
}

File: py/features.py
# LongevityCompetition SDK feature factory

from feature.base_feature import LongevityCompetitionBaseFeature
from feature.test_feature import LongevityCompetitionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LongevityCompetitionBaseFeature(),
        "test": lambda: LongevityCompetitionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
